plugins/mcreator-core/blockly/js/mcreator_extensions.js

```diff
@@ -87,6 +87,41 @@ Blockly.Extensions.registerMutator('variable_entity_input',
         }
     });
 
+// Helper function to use in Blockly extensions that register one data list selector field to update contents of another
+// The block may define input called "<targetName>Field" to customize field's position
+// Note that the source field must be inserted before the target field for their values to be loaded properly
+function appendAutoReloadingDataListField(sourceName, targetName, targetList) {
+    return function () {
+        const thisBlock = this;
+        (this.getInput(targetName + 'Field') || this.appendDummyInput()).appendField(
+            new FieldDataListSelector(targetList, undefined, {
+                'customEntryProviders': function () {
+                    return thisBlock.getFieldValue(sourceName);
+                }
+            }), targetName);
+        this.setOnChange(function (changeEvent) {
+            if (changeEvent.type === Blockly.Events.BLOCK_CHANGE &&
+                changeEvent.group && changeEvent.blockId === this.id &&
+                changeEvent.element === 'field' &&
+                changeEvent.name === sourceName) {
+                const group = Blockly.Events.getGroup();
+                Blockly.Events.setGroup(changeEvent.group);

```

Comment:
Maybe just add a short comment why this is added so we don't scratch our heads in the future (similar to comments already in similar cases in this file)

Reply:
✔️ Fixed